samples/subsys/ipc/openamp_rsc_table/src/main_remote.c

```diff
@@ -19,9 +19,14 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(openamp_rsc_table, LOG_LEVEL_DBG);
 
-#define RPMSG_CHAN_NAME	"rpmsg-client-sample"
+#define RPMSG_CHAN_NAME	"rpmsg-virtual-tty-channel-1"
 #define SHM_DEVICE_NAME	"shm"
 
+// RPMSG MU channel index
+// As Linux suggests, use MU->Data Channel 1 as communication channel
+// see RPMSG-Lite rpmsg_platform.c -> platform_notify
+#define RPMSG_MU_CHANNEL (1)
```

Comment:
This is a platform specific hack, which we can't force like this.

Reply:
I can confirm that this change is also required to successfully run the openamp_rsc_table sample on an i.MX8MP SoC.
The Cortex A53 runs Linux as the RPMSG master and Zephyr on the M7 as the remote.
I've also seen this behavior in NXP's (MCUXpresso SDK) FreeRTOS-based sample code.
Since the scenario described will probably be the standard use case, it would be nice if the Zepyhr openamp_rsc_table would also work out-of-the-box.

@pdgendt: Any suggestion how to achieve this with a mainlineable change?

Reply:
I'm not sure, maybe @arnopo can help with that?

Reply:
The best solution would be to migrate on maibox interface with declaration of maibox in TD for the Rx and tx as done in the [ipc_service sample](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/subsys/ipc/ipc_service/static_vrings/boards/nrf5340dk_nrf5340_cpuapp.overlay).
That's said, as a first step a temporary fix can be applied in the example. 
But this breaks the legacy. So i would propose:
```
if (IS_ENABLED(CONFIG_SOC_SERIES_IMX8MM_M4)
	ipm_send(ipm_handle, 0, RPMSG_MU_CHANNEL, NULL, 0);
else
	ipm_send(ipm_handle, 0, id, NULL, 0);
```

Reply:
As I still haven't figured out why this works on some platforms and not others, I can't suggest a reasonable solution.

If there really is a short-term hack for this example, then it might be better to make it dependent on CONFIG_IPM_IMX_REV2, for example, rather than start listing SOCs.

Reply:
> As I still haven't figured out why this works on some platforms and not others, I can't suggest a reasonable solution.
> 
> If there really is a short-term hack for this example, then it might be better to make it dependent on CONFIG_IPM_IMX_REV2, for example, rather than start listing SOCs.

That's valid.

To explain the context `id` parameter values are defined in the [resource table](https://github.com/zephyrproject-rtos/zephyr/blob/main/lib/open-amp/resource_table.h#L15-L17):
One Id per vring + one id for the vdev status.
Then the platform can use specific ipm channels depending on the `id`.

I have never test it but I suppose that these ID values can be all set to RPMSG_MU_CHANNEL for the IMX platform using a CONFIG or a similar mechanism.
Notice that these IDs  are also used by the NXP Linux kernel on the remote processor so can impact.

Reply:
I think it would be great for many people wanting to try Zephyr on an imx platform if the open-amp example worked out-of-the-box for that. A simple, possibly acceptable variant to realize this would be the following:

Could we extend Kconfig of openamp_rsc_table with a config like that:
```
config OPENAMP_IPC_NOTIFY_CHAN
       int "Mailbox channel to use to notify the host"
       default 0
```

And convert the code to this:
```
ipm_send(ipm_handle, 0, CONFIG_OPENAMP_IPC_NOTIFY_CHAN, NULL, 0);
```

Then any necessary adjustments to the mailbox channel used can be implemented in the board-specific configuration files (together with the device tree overlays).

For example, this line is added to mimx8mp_evk_itcm.conf
```
ONFIG_OPENAMP_IPC_NOTIFY_CHAN=1
```
In my opinion, existing and already working samples do not need to be changed and should just continue to work.